baseapp/error.go

```diff
@@ -45,7 +45,16 @@ func HandleRouteError(w http.ResponseWriter, r *http.Request, err error) {
 		Err(err).
 		Msg("Unhandled error while serving route")
 
-	WriteJSON(w, http.StatusInternalServerError, map[string]string{
-		"error": http.StatusText(http.StatusInternalServerError),
-	})
+	// Either the deadline has passed or the request was canceled
+	// 499 is an NGINX style response code for 'Client Closed Connection'
+	// and is a non-standard, but widely used, HTTP status code
+	if cerr := r.Context().Err(); cerr != nil {
```

Comment:
After thinking about this a bit, I think we should compare `cerr` to `context.Canceled` and then only return the 499 in that case. It would also be good to remove the error logging in this case (or perhaps log a special message at a lower level).